app/routes.js

```diff
@@ -57,18 +61,18 @@ exports.bind = function (app) {
 
   app.get(card.auth3dsRequired.path, middlewareStack, threeDS.auth3dsRequired)
   app.get(card.auth3dsRequiredOut.path, middlewareStack, threeDS.auth3dsRequiredOut)
-  app.post(card.auth3dsRequiredInEpdq.path, [retrieveCharge, resolveLanguage], threeDS.auth3dsRequiredInEpdq)
-  app.get(card.auth3dsRequiredInEpdq.path, [retrieveCharge, resolveLanguage], threeDS.auth3dsRequiredInEpdq)
-  app.post(card.auth3dsRequiredIn.path, [retrieveCharge, resolveLanguage], threeDS.auth3dsRequiredIn)
-  app.get(card.auth3dsRequiredIn.path, [retrieveCharge, resolveLanguage], threeDS.auth3dsRequiredIn)
-  app.post(card.auth3dsHandler.path, [actionName, retrieveCharge, resolveLanguage, resolveService, stateEnforcer], threeDS.auth3dsHandler)
+  app.post(card.auth3dsRequiredInEpdq.path, middlewareStackFor3dsWithSessionValidation, threeDS.auth3dsRequiredInEpdq)
+  app.get(card.auth3dsRequiredInEpdq.path, middlewareStackFor3dsWithSessionValidation, threeDS.auth3dsRequiredInEpdq)
+  app.post(card.auth3dsRequiredIn.path, middlewareStackFor3dsWithSessionValidation, threeDS.auth3dsRequiredIn)
+  app.get(card.auth3dsRequiredIn.path, middlewareStackFor3dsWithSessionValidation, threeDS.auth3dsRequiredIn)
+  app.post(card.auth3dsHandler.path, [actionName, middlewareStackFor3dsWithSessionValidation, resolveService, stateEnforcer], threeDS.auth3dsHandler)
 
   // Apple Pay endpoints
   app.post(paths.applePay.session.path, applePayMerchantValidation)
 
   // Generic Web payments endpoint
-  app.post(paths.webPayments.authRequest.path, retrieveCharge, resolveLanguage, webPaymentsMakePayment)
-  app.get(paths.webPayments.handlePaymentResponse.path, retrieveCharge, resolveLanguage, webPaymentsHandlePaymentResponse)
+  app.post(paths.webPayments.authRequest.path, middlewareStackFor3dsWithSessionValidation, webPaymentsMakePayment)
```

Comment:
The naming doesn’t seem right if we’re going to apply this to web payments stuff

Reply:
What do you think about middlewareStackWithSessionValidation, that will mean the next one would be middlewareStackWithoutSessionValidation.

Any suggestions would be good.

Reply:
I’m a bit conflicted on this. I’m not sure if we should be naming based on what the middleware does (which runs the risk of getting out of date when we decide to add something new to a middleware stack) or the sorts of pages it applies to (which runs into problems when you decide your 3DS middleware would be ideal for something non-3DS).

Right now we have `middlewareStack`, which is used for most pages, so maybe that could become `standardMiddlewareStack`. And this new one could be `chargeCookieRequiredMiddlewareStack` (because that’s basically all it does). And the one that doesn’t validate the session could be `chargeCookieNotRequiredMiddlewareStack`. Perhaps not the most future-proof but we can easily rename them if our needs change.

Reply:
I understand that the names can become out date, but are simply trying to make the code more readable. Hopefully when changes are made in the future they will also give the naming the same due diligence.

I will go with your suggestions for now.